### ISSUE TITLE
Moving the project away from github

### ISSUE THREAD

**Issue by ghost** [mpv-player/mpv#7924]
I think it's time for this project to look for another host.

Github is now owned by Microsoft. Despite trying to look cool and flirting with open source etc., Microsoft is still very much evil. We don't need even to discuss this: just consider their past or present. For example, in the past they've spread massive FUD against Linux and free software, and currently they're trying the opposite strategy by EEEing open source / free software (the github purchase is an example). They're also a bad influence on society as a whole, for example consider them [helping governments to implement the surveillance state](https://theintercept.com/2020/07/14/microsoft-police-state-mass-surveillance-facial-recognition/).

The minimum we need is a git host and some sort of issue tracker. I'd be fine with a raw git server, and coordinating development on IRC only, but I suspect others wouldn't want to do without at least some web-based integrated issue tracker.

Suggestions welcome.

**Comment by haasn** [comment]
I have conflicting thoughts on this.

1. How is Microsoft's ownership of GitHub actually affecting the mpv project? Is this about mpv needing to migrate, or is this merely about you wanting to somehow "hurt" Microsoft by moving mpv away from GitHub? Can you provide technical justification for this or is it a merely political move?

2. I would very much advise against trying to go 'bare' using IRC/MLs/whatever as the only means of coordination. That might work for you alone but nobody else, and mpv is still a community project.

3. I would prefer using something that has an established userbase, especially in the multimedia sector. I've been very happy with [code.videolan.org](https://code.videolan.org/).

**Comment by CounterPillow** [comment]
Doesn't mpv.io also use GitHub pages?

**Comment by wm4** [comment]
> How is Microsoft's ownership of GitHub actually affecting the mpv project? Is this about mpv needing to migrate, or is this merely about you wanting to somehow hurt Microsoft by moving mpv away from GitHub? Can you provide technical justification for this or is it a merely political move?

There is no technical need. For one, we should obviously not support something that will destroy us. Of course, our actions will have no consequences at all for Microsoft. But it's a moral obligation, and other open source projects (which are "still" hosted elsewhere) explicit choose not to use github. The more projects abandon github, the better, and we may as well be part of the (probably very small) stream of projects emigrating from github.

Another reason might be that it's not unlikely that github may impose random restrictions on its users. Consider the recent [GNOME incident](https://old.reddit.com/r/gnome/comments/hn1s3r/mpv_is_not_anymore_supporting_gnome_and_the_owner/fxatwj9/) (nothing happened, but shows that the more they go off the deep end, the more "dangerous" it gets), or [getting randomly blocked by github for no reason at all.](https://news.ycombinator.com/item?id=23832437) That might count as FUD towards github, but I think it's effective.

> Doesn't mpv.io also use GitHub pages?

Yes.

**Comment by CounterPillow** [comment]
GNOME's attempted sabotage of the mpv GitHub repository is worrisome, but that GitHub did not act on a clear attempt of malicious interference by a toxic bully is reassuring.

As for the HN thing, my experience is that the one time my GitHub account got limited for posting a gist that had one too many URLs in it or something like that, the account limitation/"ban" was fairly easy to get lifted, and in general GitHub support has been fairly responsive for me.

code.videolan.org seems like a good new home if the decision really is made to move, partly because of the "VLC bought out mpv" shitposts people will make but also because of the platform's focus on multimedia projects and VideoLAN's long history of commitment towards free and open multimedia. Though I still think this is a lot of effort/disruption for a move that is mainly ideology/FUD driven.

*Full disclosure: GitHub is currently paying me through matched sponsorships*

**Comment by haasn** [comment]
> Consider the recent GNOME incident (nothing happened, but shows that the more they go off the deep end, the more "dangerous" it gets)

I would personally take the fact that nothing happened as more of a sign *against* needing to move, rather than as a sign needing to move. Arguably, moving as a precautionary measure vs moving when actually forced to move makes no real difference in terms of inconvenience caused.

But nonetheless, I understand your concerns fully; and do agree that decentralization is better than centralization as a general principle; so moving onto a platform we control more or less directly (as far as I'm concerned, I consider 'code.videolan.org' to be 'our own infrastructure') should be a good move independent of any politics involved.

I think I'll summarize my position as a 'weak agree under the condition that whatever we migrate to won't be a functionality downgrade'. But I'm in no hurry to move.

Also, I think that you should anticipate the need to keep around an official mirror on GitHub, and maybe even enable the issue tracker there. (That's what I do for [libplacebo](https://github.com/haasn/libplacebo))

**Comment by Akemi** [comment]
since i am a lazy bum, i would rather not move somewhere else. though i won't be in the way if we should decide to. for the same reason i would be kinda against managing a mirror or rather two separate issue trackers.

i guess code.videolan.org is probably the best alternative for us.

**Comment by orbea** [comment]
I just want to point out that last time I created an issue for code.videolan.org I was unable to later reply to the issue because I couldn't easily get 2fa to work...

**Comment by Dudemanguy** [comment]
How would moving to code.videolan.org work? Would we have to follow this [Code of Conduct](https://wiki.videolan.org/Code_of_Conduct/)? If so, it seems a pretty clear no go to me since "No insult in commits or commit messages." has been violated many times over many years (and personally I would like to keep it that way because it's more fun).

**Comment by philipl** [comment]
I don't put any weight into the conspiracy theories about Microsoft. They are a very different company from what they were 15-20 years ago, and it's self-defeating for us to base our decisions on how they used to behave. You can certainly have a discussion of the technical merits of github vs gitlab vs videolan hosted gitlab etc, but from my experience of all three, I don't see any benefits that justify the costs of moving.

**Comment by sfan5** [comment]
I value the convenience of having all projects that I contribute to on the same platform, so I'd rather not move platforms solely for the moral/political argument and wait until there is a concrete indicator that Microsoft wants to go the "wrong" way with Github (@haasn's comment pretty much covers my opinion regarding this).
Microsoft owns Github since late 2018 and so far hasn't done anything unfriendly to FOSS projects in general. 

Also, although Gitlab is probably the best alternative of all I strongly dislike the heavy use of Javascript all over the place. You download a page to view a source file and yet have to stare at a loading icon for one or two seconds (actual measurements) until content actually appears.

**Comment by olifre** [comment]
I also agree with @haasn's and @sfan5's comments here. 
As many large companies, Microsoft seems to have discovered the value of open source in recent years, and unlike some other big companies, they seem not to have set out to destroy FOSS products as of yet, but even have some FOSS projects of their own. I won't praise any of these products, but there does not seem to be any clear unfriendly behaviour from them as of now. 

Of course, nobody can predict the future, but waiting until they give us a real reason instead of hastening the decision seems reasonable to me. In any case, I agree migrating is not a low amount of work (e.g. it might not be a bad thing to keep archived issues, for example, if only to point users to them to be able to say "we've already answered that"), and it should not be a functional downgrade.

**Comment by wm4** [comment]
https://en.wikipedia.org/wiki/Embrace,_extend_and_extinguish

Learn from history.

**Comment by barolo** [comment]
@olifre wait till you get hit in the face then make a decision, great advice mate...
They don't care about you, they will throw an army of lawyers at you without blinking if you overstep their perceived boundaries.
Microsoft is "embracing open-source" only because they can't compete in some fields. Just... learn from the history, not only an IT one.   
Everything is politics nowadays, it's not if you're ignorant or you aren't looking hard enough, global reach of everything ensures that.  I wouldn't act on it alone but let's not use that as argument against

**Comment by haasn** [comment]
> @olifre wait till you get hit in the face then make a decision, great advice mate...

What do you expect to have changed about our ability to move the mpv codebase elsewhere after we've gotten 'hit in the face'?

**Comment by barolo** [comment]
> > @olifre wait till you get hit in the face then make a decision, great advice mate...
> 
> What do you expect to have changed about our ability to move the mpv codebase elsewhere after we've gotten 'hit in the face'?

you're trying to say that you want to "get hit in the face" [ by a well known bully ] first, then make your move?

**Comment by haasn** [comment]
> are you trying to say that you want to "get hit in the face" [ by a well known bully ] first, then make your move?

Yes, that's exactly what I said in my previous post on this matter.

**Comment by orbea** [comment]
> What do you expect to have changed about our ability to move the mpv codebase elsewhere after we've gotten 'hit in the face'?

I think the argument is that a planned move would be easier than an unexpected and unplanned move if and when github does something that would require that.

**Comment by olifre** [comment]
> https://en.wikipedia.org/wiki/Embrace,_extend_and_extinguish

Interesting! Apparently I missed most of that, since my usage of Microsoft products is too low (basically nothing apart from GitHub). At least, they also lost some of these wars, e.g. the browser war, no matter how hard they tried on this end. 

However, I also see that all the examples of the past had a lock-in possibility designed into the product, and customers still actively deciding for it. I don't see how they can take the code from us, or otherwise lock us in on GitHub, or in any way take away the possibility to move away at a later point. 

So even after catching up on history, I'll go with @haasn and agree weakly, with no need to hurry from my point of view. Indeed, I'd love to see them hitting me in the face.

**Comment by barolo** [comment]
> > are you trying to say that you want to "get hit in the face" [ by a well known bully ] first, then make your move?
> 
> Yes, that's exactly what I said in my previous post on this matter.

different strokes for different folks, I guess... I'd rather not get hit at all

**Comment by haasn** [comment]
> I think the argument is that a planned move would be easier than an unexpected and unplanned move if and when github does something that would require that.

Conversely, an unexpected and unplanned mass exodus would certainly draw far greater attention to the blow that triggered it; and thus, at the very least, be more amusing.

Anyway, I guess the main benefit to moving now would be that we have time to leave behind a trail of breadcrumbs leading to the new location *before* the `mpv` repository gets permanently deleted, in that hypothetical circumstance. Not that I expect that to ever happen. It doesn't even make sense from the 'EEE' perspective. Nowhere in that article does it mention how Microsoft turned into a corrupt dictator shutting down people's computers left and right. **What 'extinguish' means here is that *they* want *us* to continue using GitHub instead of alternatives,** by making GitHub so much more integrated that all alternatives cease to provide the same level of convenience/compatibility.

So if anything, the extent of what we have to worry about is them trying to make our workflow depend on the use of their proprietary tools. What can do to stave off fears of EEE is avoiding depending on anything that we couldn't easily replace by a different tool if need be. And as far as I'm concerned, the only things we use are issues (provided everywhere), pull requests (provided everywhere), and the wiki (provided everywhere).

> different strokes for different folks, I guess... I'd rather not get hit at all

Well, you're welcome to stop using GitHub. Personally, I don't let anxiety and paranoia run my life.

**Comment by barolo** [comment]
> Well, you're welcome to stop using GitHub. Personally, I don't let anxiety and paranoia run my life.

The trusting ones get murdered first.  
Joking aside I agree with you in parts, I like to look at thunderstorms and bushfires, drama is entertaining.
My own take is that I just don't like corpos, shady corpos and I'd rather not support them by using their platform.

**Comment by haasn** [comment]
> How would moving to code.videolan.org work? Would we have to follow this [Code of Conduct](https://wiki.videolan.org/Code_of_Conduct/)? If so, it seems a pretty clear no go to me since "No insult in commits or commit messages." has been violated many times over the years (and personally I would like to keep it that way because it's more fun).

No. The Code of Conduct you linked applies to VideoLAN projects. Nothing is stopping individual users from registering on c.v.o and uploading their own projects, and merely being hosted on c.v.o does not make something a VideoLAN project.

However, as the server is hosted in France, you will have to comply with French law. While, I'm not familiar with French hate speech laws, I'm going to strongly assume that you can continue referring to GNOME/Wayland/dbus/unicode/whatever as 'a pile of radioactive monkey feces'.

**Comment by bitsper2nd** [comment]
Github still works great the mpv project. I see no hurry to change to another code hosting and review service. I understand @wm4 and @haasn points. So I suggest on studying the alternatives first. There are more options than just github or gitlab and do the test on a fork and other smaller projects as a team to see if its viable. Github's biggest advantage is the social lock in effect. Since there are many people already on the platform so it's easier to get more eyes and possible collaborator. It's also easier to get more nagging(such is life.) So it's a question of priorities of the project and the value of the tool, do you value having more eyeballs and possible collaborators in the mpv project? Is the alternative feature par with what Github offers? Are the members of the project comfortable with the new tool? I don't see you guys switching to something like Drew DeVault's Sourcehut for example. There is no is rush, you guys definitely have time to figure things out before the unexpected punch comes in.

**Comment by wm4** [comment]
FDO's gitlab instance might also work: https://gitlab.freedesktop.org

**Comment by wm4** [comment]
Also plus points if the target platform supports github import: wiki, issues, PRs.

**Comment by haasn** [comment]
My experience with GitLab was that it supports importing projects from GitHub pretty much automatically (including mapping user accounts for stuff like issues, assuming users are registered on both)

**Comment by bitsper2nd** [comment]
Here is a list of features found in [Codeberg](https://docs.gitea.io/en-us/comparison/), which uses Gitea. The Gadgetbridge Team imported their old github issues when they migrated.

**Comment by Comevius** [comment]
RIP mvp.

https://github.com/mpv-player/mpv/commit/fe791de43e9bd33711e5bc9b6ed360cccd22acb1
https://github.com/mpv-player/mpv/commit/269f0e743e5634691f0c9d5b1b8a4bb68eedbbd0

**Comment by q3cpma** [comment]
As a user, I really recommend Sourcehut, even if it is somewhat in a (fast evolving) beta state. Switching to a jabbascript pile of shit like Gitlab would be a net loss compared to Github, please don't.

**Comment by keithbowes** [comment]
M$ bought Github ages ago, so why just bring this up now?  I momentarily moved to Gitlab, but its functionality was just too lacking; plus, I had to keep my Github account to submit issues, PRs, etc to the FOSS projects on Github, which is most of them (I experience more projects hosted on SourceForge than Gitlab).

**Comment by WizRay** [comment]
@keithbowes, find the reason [here](https://github.com/mpv-player/mpv/issues/7924#issuecomment-659300271).

**Comment by easytarget** [comment]
https://codeberg.org/ provides tracking free & free ( (as in beer) ) hosting using [GitTea](https://gitea.io/en-us/).

I've been running GitTea (formally Gogs) for years on my personal server, and like it a lot; it has developed enormously in that time and is now, basically, a fully featured github clone now. Quick demos at: https://try.gitea.io/, 
It has import facilities, not too sure how good these are from github, but people have [made the switch](https://dev.to/jmarhee/migrating-from-github-to-gitea-26ei) successfully..

**Comment by balibebas** [comment]
I don't personally understand how there can even be any arguments in favor of GH here. This is GPL software. Ethics are baked into the license so not moving is complacency and that runs counter to the ethos free software tries to adhere to. If you need a reason to hate Microsoft check out the details about building nuclear weapons [for their enterprise users](https://habd.as/post/coding-mass-destruction/).

If you want free hosting with issues why not use what the FSF recommends in [GNU Ethical Repository Criteria Evaluations](https://www.gnu.org/software/repo-criteria-evaluation.html). Or you could also use [gitea.com](https://gitea.com) because that's a thing now too and it's way faster than these privacy rent-seeking platforms like GH/GL.

**Comment by garoto** [comment]
> If you need a reason to hate Microsoft check out the details about building nuclear weapons [for their enterprise users](https://habd.as/post/coding-mass-destruction/).

Hating Microsoft is easy, all it takes is using any Windows  (or DOS) version **ever** marketed to be able to comprehend their full blown commitment towards shitness, no need to bring nuclear weapons into this.

**Comment by mlindner** [comment]
> Despite trying to look cool and flirting with open source etc., Microsoft is still very much evil. 

[citation needed]

How about we not bandwagon?

**Comment by mlindner** [comment]
> https://en.wikipedia.org/wiki/Embrace,_extend_and_extinguish
> 
> Learn from history.

So you're saying company culture never changes?

**Comment by haasn** [comment]
> If you need a reason to hate Microsoft check out the details about building nuclear weapons [for their enterprise users](https://habd.as/post/coding-mass-destruction/).

I'm confused. Somebody explain this to me. I'm supposed to hate Microsoft because they **forbid the use of their product to develop nuclear weapons**?

Is this some extremely twisted sense of "not being allowed to develop nuclear weapons is a violation of freedom zero!" argument, or what am I missing here?

**Comment by balibebas** [comment]
> Is this some extremely hyperbolic "not being allowed to develop nuclear weapons is a violation of freedom zero!" argument, or what am I missing here?

They don't forbid the use of GitHub to build WMDs. They forbid the use of GitHub to build WMDs unless you're an American company. If there was a soul in the business they would simply forbid it for everyone.

**Comment by balibebas** [comment]
> Is this some extremely hyperbolic "not being allowed to develop nuclear weapons is a violation of freedom zero!" argument, or what am I missing here?

They don't forbid the use of GitHub to build WMDs. They forbid the use of GitHub to build WMDs unless you're an American company. If there was a soul in the business, they would forbid it for everyone.

**Comment by balibebas** [comment]
> Is this some extremely hyperbolic "not being allowed to develop nuclear weapons is a violation of freedom zero!" argument, or what am I missing here?

They don't forbid the use of GitHub to build WMDs. They forbid the use of GitHub to build WMDs unless you have American approval. If there was a soul in the business, they would forbid it for everyone.

**Comment by balibebas** [comment]
> Is this some extremely hyperbolic "not being allowed to develop nuclear weapons is a violation of freedom zero!" argument, or what am I missing here?

They don't forbid the use of GitHub to build WMDs. They forbid the use of GitHub to build WMDs unless you have American approval. If there was a soul in the business, they would forbid it for everyone regardless.

**Comment by balibebas** [comment]
> Is this some extremely hyperbolic "not being allowed to develop nuclear weapons is a violation of freedom zero!" argument, or what am I missing here?

They don't forbid the use of GitHub to build WMDs. They forbid the use of GitHub to build WMDs unless you have American approval. If there was a soul in the business, they would forbid it for everyone regardless.

To avoid strawman please address the stronger argument. And that's the FSF ethics link I attached above.

**Comment by Argon-** [comment]
They need to do this, this is not their choice. Those are rules mandated by US export laws. 
This is the same when you buy a super computer from Cray for use outside of the US. Simply because they could be used to simulate nuclear detonations, a report must be filed to US authorities for every single computation you run on "your" super computer.

**Comment by wm4** [comment]
> So you're saying company culture never changes?

Generally, it gets worse. While you're being fooled by MS trying to look cool, they still do plenty of evil things, such as making hardware on which only an OS signed by MS can run.

**Comment by wm4** [comment]
Another argument: https://github.com/github/roadmap/issues/63
Stuff like this is going to get worse, because it's an US company.

**Comment by mlindner** [comment]
Well if you're not interested in having an open source project that many contributors can easily view and only interested in having a project that you personally commit to, then sure put it anywhere.

**Comment by JC85206** [comment]
@mlindner 
![att](https://user-images.githubusercontent.com/68624816/88821179-dddebc80-d1b1-11ea-915c-aac911634246.jpg)

**Comment by wm4** [comment]
I also don't want to participate in helping monopolies out of convenience.

**Comment by bitsper2nd** [comment]
> I also don't want to participate in helping monopolies out of convenience.

I understand this sentiment. The problem is that the situation is out of your hands. Best you can do is switch to another git hosting provider and change nothing in the general landscape. Microsoft is already a monopoly and the US government is the one that let's it keep buying other companies such as Github. The sad truth is that it's up to Americans and their leaders to put a stop to corporations like Microsoft.

**Comment by wm4** [comment]
> The problem is that the situation is out of your hands. Best you can do is switch to another git hosting provider and change nothing in the general landscape.

No, but see https://github.com/mpv-player/mpv/issues/7924#issuecomment-659300271. If nobody ever tried to do the right thing, nothing would ever improve.

> Microsoft is already a monopoly and the US government is the one that let's it keep buying other companies such as Github. The sad truth is that it's up to Americans and their leaders to put a stop to corporations like Microsoft.

Well sure, feel free to lobby to american politicians or something.

**Comment by ritiek** [comment]
There's also this [ongoing discussion on reddit](https://www.reddit.com/r/opensource/comments/i39107/open_source_developers_should_abandon_github_imo/) if someone cares.

**Comment by emanuelserpa** [comment]
Embrace: https://www.phoronix.com/scan.php?page=news_item&px=Microsoft-WSL2-Coming
Extend: https://www.phoronix.com/scan.php?page=news_item&px=Microsoft-DX12-WSL2
Extinguish: ???

**Comment by selsta** [comment]
Travis-CI (currently used by mpv) only supports Github and Bitbucket.

**Comment by easytarget** [comment]
> Embrace: https://www.phoronix.com/scan.php?page=news_item&px=Microsoft-WSL2-Coming
> Extend: https://www.phoronix.com/scan.php?page=news_item&px=Microsoft-DX12-WSL2
> Extinguish: ???

I've been thinking about this. Too much time on my hands (thanks to Covid).
* They Embrace; but this means a lot of their code becomes open source (to varying degrees, depending on licence). It also means they get a lot of hard (software, data and specification) work 'for free', saving money and achieving compatibility, but paying some 'cost' in terms of having to hand a lot of code and protocol data back to the community.
* They Extend; In ways that favour them and not their competitors, occasionally in new or unexpected ways. But the same constraints as above make them share a lot of this back to the community, so that it becomes part of the ecosystem. 
* My struggle is with the idea that they can simply 'extinguish' code that is widely shared. 
The best routes are to socially engineer it almost out of existence, combined with working in the legislative sphere to generate an idea that the code is now 'theirs' and others must be stopped from seeing it. Fostering a spirit in the OSS movement of infighting and storming off in a hissy fit could also be part of this. Just saying...
My feeling is that the more a company supports authoritarianism (in all forms, social and technical) the more you should be wary of them. I'd be looking more askance at Facebook (another OSS embracing+extending company) slightly more than MS in this regard. While trying to keep another eye on the varied companies that have slowly lost their ethics after gaining shareholders, many of whom see profit and power in authority.

**Comment by emanuelserpa** [comment]
@easytarget

I'm too sleepy to give you a proper answer but this DirectX thing they are doing for Linux is just a pipeline to DirectX use the native drivers for Windows in the containerized Linux guest (WSL2) and they are doing a custom Wayland compositor too (that can be proprietary, btw) to work around that limitation too.

They are doing that without returning the favor (allowing Linux native graphics work in Windows guests, for example).

While what they are doing will not properly "extinguish" Linux they are trying to create a Windows-only ecosystem for WSL2 (**Extend:** _Addition and promotion of features not supported by the competing product or part of the standard, creating interoperability problems for customers who try to use the "simple" standard_), and soon we can see apps that nominally are for Linux that will not work in baremetal Linux, only in WSL2.

In conclusion, I see that everything they are doing now is to get a platform Lock-in, cementing Windows in Linux user cases in the industry (like workstations) and cement baremetal Linux as simply a hobbyOS without any usercases outside webservers(render farms and machine learning).

**Comment by easytarget** [comment]
@emanuelserpa 
You are not wrong, I just think you fear MS more than is warranted. 

I was a *nix then linux sysadmin in the late 90's & early 00's, and it felt worse back then. I couldn't even send emails to many Outlook recipients (aka customers), with their IT departments blaming me for using smtp instead of an exchange server. At one time I was so low that was wondering if I should take a MS qualifications, or become a vet. But, with 20 years hindsight it actually worked out better in the end; and I get to type this on a very modern netbook that is running Fedora and is 100% functional both as a development machine and as a desktop. As for issues with linux Video drivers; plus ça change. 

I do see and understand what you mean about the WSL stuff, and that is doubtless how some of the business peons at MS think it will go. They tried the same thing when they 'extended' html to make IE6, and toppled Netscape, but by killing them they cleared the way for Firefox then Chrome. I can see competitors and community working around it and WSL is likely to end up as a 'ms standards only' fork that costs them a lot to maintain for an ever dwindling client base. Hopefully.

Also; they chose Wayland. HARHARHAR foolish MS. The Wayland devs and codebase are even worse then the Gnome ones when it comes to the 'I flat refuse to consider your use case' school of complexity reduction. Resulting in chunks of missing functionality. The only way to get progress on Wayland is to fork it, or spend a fortune redoing from scratch.

**Comment by thebombzen** [comment]
It's worth mentioning that it's not as if all the code will irreparably disappear if GitHub suddenly randomly shuts down MPV, without warning. We all have local clones of the repository. Would the issue tracker's content be saved? No, but that would also be very difficult to migrate should we choose to move somewhere else voluntarily.

**Comment by wm4** [comment]
> My struggle is with the idea that they can simply 'extinguish' code that is widely shared.

Saying that EEE will happen with github is probably misleading (why would they extinguish their own site), but it's still a good reminder of MS company culture.

But this doesn't mean they will always have a "good" influence. Their main goal (since they're a company) is to make people dependent on their products and services. They've purchased github, so there's no need to "extinguish" it anymore, but a sourceforge situation might arrive later if MS ever thinks it need to squeeze our some money. They also don't need to extinguish Linux anymore, but they'll just make people depend on WSL - they don't need to care about the fact that there's still Linux, as long as other companies need Windows to use it, and pay for their Windows license. (Maybe making companies to require crazy Windows/Linux hybrid deployments instead of Linux only is their wet dream?)

Then there's still the possibility that MS takes over control over open source projects themselves, like it happens all the time to such projects if companies get "invested" in them. Maybe KHTML->WebKit can serve as an example.

Currently, Microsoft tries to appear "cool" and will avoid any kind of offensive moves. If you actually believe them, just look at how they treat their Windows 10 users to see the corporation instead of the "cool". Maybe you consider that FUD, but then consider which company used FUD the most as marketing strategy. Penguins with antlers, I say.

**Comment by wm4** [comment]
Funny look at the past: http://techrights.org/2015/01/13/gnu-gpl-still-dominant/

**Comment by G2G2G2G** [comment]
> > https://en.wikipedia.org/wiki/Embrace,_extend_and_extinguish
> > Learn from history.
> 
> So you're saying company culture never changes?

Huh? They change all the time, companies constantly try to screw everyone over, even their own employees. If they're publicly ran there's a guarantee they hate you and everyone involved because they have to give them a paycheck and there's nothing worse. Github is closed source abomination of git and it always has been. There's no reason to use this site, people just want their coding social media.

**Comment by v-fox** [comment]
>> different strokes for different folks, I guess... I'd rather not get hit at all
>
>Well, you're welcome to stop using GitHub. Personally, I don't let anxiety and paranoia run my life.

This did not age well after [mass-banning people](https://github.com/1995parham/github-do-not-ban-us) on basis of their citizenship by [the order of US government](https://techcrunch.com/2019/07/29/github-ban-sanctioned-countries/) (I wonder when I'm going to get banned with all the Russians and not just some of us, like now) and [removing youtube-dl](https://github.com/mpv-player/mpv/issues/8201) (while sabotaging its unmerged code from pull/merge request and issue database) by [RIAA](https://en.wikipedia.org/wiki/Recording_Industry_Association_of_America) who is a good buddy with [MPEG LA](https://en.wikipedia.org/wiki/MPEG_LA), THE anti-multimedia patent troll championed by Microsoft.

As was said previously, FDO or VLC GitLab instances seems like a straightforward and reliable migration paths. FDO would be particularly good place for mpv, especially if it's going to [embrace pipewire](https://github.com/mpv-player/mpv/pull/7902) that is also [hosted there](https://gitlab.freedesktop.org/pipewire/pipewire/) along with [Mesa](https://gitlab.freedesktop.org/mesa/mesa/) and [Wayland](https://gitlab.freedesktop.org/wayland). Would be nice to somehow coax ffmpeg and youtube-dl into getting together in the move with mpv too.

> Another argument: github/roadmap#63
> Stuff like this is going to get worse, because it's an US company.

However, FDO seems to be getting down with American "liberal" language policing of "political correctness" (correct only from the point of view of the American "liberals", of course) via its [Code of Conduct](https://www.freedesktop.org/wiki/CodeOfConduct/). They even started Holy Inquisition against "blacklists". I don't know if that's entirely on project owners or the entire repo but seems to be the latter.

> They need to do this, this is not their choice. Those are rules mandated by US export control laws.
> This is the same when you buy a super computer from Cray for use outside of the US. Simply because they could be used to simulate nuclear detonations, a report must be filed to US authorities for every single computation you run on "your" super computer.

Exactly ! All development-related meta-data is not part of git and is not saved together with the repo mirrors. Everything here tied not just to software but to its hoster who is an US resident under jurisdiction of its pseudo-democratic political theatre of the ruling oligarchy.

I have good friends who work in area of non-US nuclear weapons and delivery systems, I endource them to adopt F/OSS in their workspace and personal devices. Russian & Chinese nuclear weapons keep American ones in check, as per [the MAD principle](https://en.wikipedia.org/wiki/Mutual_assured_destruction), perfectly illustrated in [Dr. Strangelove](https://en.wikipedia.org/wiki/Dr._Strangelove) (and [some real incidents](https://en.wikipedia.org/wiki/U.S._nuclear_weapons_in_Japan#Nuclear_weapons_accidents)). That's what we get for continuing to live under psychopathic oligarchies but it's better than being fried by one of them that's decided that it's the hottest $h1t around.

**Comment by Akemi** [comment]
> I have good friends who work in area of non-US nuclear weapons and delivery systems, I endource them to adopt F/OSS in their workspace and personal devices. Russian & Chinese nuclear weapons keep American ones in check, as per [the MAD principle](https://en.wikipedia.org/wiki/Mutual_assured_destruction), perfectly illustrated in [Dr. Strangelove](https://en.wikipedia.org/wiki/Dr._Strangelove) (and [some real incidents](https://en.wikipedia.org/wiki/U.S._nuclear_weapons_in_Japan#Nuclear_weapons_accidents)). That's what we get for continuing to live under psychopathic oligarchies but it's better than being fried by one of them that's decided that it's the hottest $h1t around.

please keep such topics out of here and move such discussions to an appropriate place. imo it has no place in here and is completely irrelevant to the topic at hand. i also don't want this issue to become a fighting ground for such things.

**Comment by v-fox** [comment]
> please keep such topics out of here and move such discussions to an appropriate place. imo it has no place in here and is completely irrelevant to the topic at hand. i also don't want this issue to become a fighting ground for such things.

Haven't seem to be an issue for months previously when it was brought up and discussed :\ Unless you want to imply that "this is apolitical" which it never is because you also act on political beliefs and the issue at hand concerns political threats.
Besides, it is relevant to the topic of enforcement of US jurisdiction as some kind of default universal moral and legal obligation, for which [nuke-building is no exception](https://github.com/github/site-policy/blob/b059da293f92c65ab7ea1e683825bd9bac2032d1/Policies/github-enterprise-subscription-agreement.md) because [nothing is](https://github.com/github/site-policy/search?q=sanctions). This whole debate has started with almost a disparagement of author's "paranoia". Well, it only may seem as paranoia until you're the one sitting on a side of where a stick points.

That also pertains to one of the main drawbacks for potential alternative hostings: US-centric CoCs, like one from FDO. And I call them "US-centric" because they are based on dogma of beliefs from those supposedly liberal (which I deny and view as pacifier made-up to uphold the false dichotomy of "left" and "right" that in reality is the same blob of classist interests, AKA "oligarchs") US political movements while CoC's enforcement mirrors other political behaviour: so called "[cancel culture](https://en.wikipedia.org/wiki/Online_shaming#Call-outs_and_cancellation)" based on "[paradox of tolerance](https://en.wikipedia.org/wiki/Paradox_of_tolerance)" premise. Which makes them part of the same whole. But, then again, EU seems to have adopted it too.

In short, such CoCs suck and it would be nice to avoid them on another platform. US state enforcement sucks even more.

**Comment by Akemi** [comment]
i locked the conversation for now. everything has been said so far and we will make a decision eventually.

**Comment by kasper93** [comment]
There are currently no ongoing plans to move away from GitHub, should any such plan be formalized this issue can be reopened.
